src/schema/README.md

```diff
@@ -87,371 +400,197 @@ reference__eeg:
+    | `unit`         | Interpretation of values in a data file with the given suffix  |
+    | `maxValue`     | Maximum permissible value in a data file with the given suffix |
+    | `minValue`     | Minimum permissible value in a data file with the given suffix |
+    | `anyOf`        | Used to describe multiple permissible units                    |
+
+-   `objects.extensions`
+    | Field          | Description                 |
+    | -------------- | --------------------------- |
+    | `display_name` | Human-friendly name         |
+    | `description`  | Term definition             |
+    | `value`        | String value of `extension` |
+
+-   `objects.formats`
+    | Field          | Description                        |
+    | -------------- | ---------------------------------- |
+    | `display_name` | Human-friendly name                |
+    | `description`  | Term definition                    |
+    | `pattern`      | Regular expression defining format |
+
+-   `objects.associated_data`
+    | Field          | Description         |
+    | -------------- | ------------------- |
+    | `display_name` | Human-friendly name |
+    | `description`  | Term definition     |
+
+-   `objects.top_level_files`
+    | Field          | Description         |
+    | -------------- | ------------------- |
+    | `display_name` | Human-friendly name |
+    | `description`  | Term definition     |
 
--   `string`: If `type` is `string`, then there MAY be any of the following
-    fields at the same level as `type`: `pattern`, `format`, and `enum`.
+## Rule files
 
-    -   `pattern` defines a regular expression for valid string names.
+The `rules.*` namespace contains most of the validatable content of the schema,
+apart from value constraints that can be encoded in `objects`.
 
-    -   `format` defines the format for the string at the same level as `type`.
-        Valid values for `format` include:
+There are several types of rule, and this section is subject to reconsolidation as
+patterns are found.
 
-       -   `uri` (uniform resource identifiers),
+### Core concepts
 
-       -   `bids_uri` (BIDS URIs),
+Core concepts are [expressions](#expressions) (defined above), requirement levels and issues.
 
-       -   `date` (date-times),
+#### Requirement levels and severity
 
-       -   `unit` (standard units),
+BIDS follows RFC 2119 and has three requirement levels: OPTIONAL, RECOMMENDED and REQUIRED.
+In the schema, we use `optional`, `recommended` and `required`. A rule interpreter (validator)
+is expected to treat missing REQUIRED data/metadata as an error, missing RECOMMENDED
+data/metadata as a warning, and silently pass over missing OPTIONAL data.
 
-       -   `dataset_relative` (relative paths from dataset root),
+BIDS also defines a level `DEPRECATED`, rendered in the schema as `deprecated`, and corresponding
+to a warning if the data/metadata is present.
 
```

Comment:
It does not exist so far. I'd like to avoid it until it becomes definitely needed.

Reply:
Can't that case be covered with a `!(ImProhibited in blah)`?

Reply:
That would be a check, not a requirement level. So yes, it can be done so far. Maybe there will be a case where we'd have to bend over backwards to avoid it and it makes more sense to go ahead. Haven't seen it yet.

Reply:
@rwblair maintains the validator, so I could go either way on this :)

Reply:
There's a pr some where with prohibited as a requirement level, I think its useful for the mutual exclusion cases.